extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -185,52 +179,46 @@ public void run() {
-                                    .build();
-                            jobDetail = newJob(getDailyJobClass()).withIdentity("job-daily-startup-" + typeId, thingUid)
-                                    .usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
+                            Calendar start = Calendar.getInstance();
+                            start.add(Calendar.SECOND, thingConfig.getInterval());
+                            DateExpression dateExpression = new DateExpression(
+                                    AbstractBaseJob.ISO8601_FORMAT.format(start.getTime()));
+                            getScheduler().schedule(
+                                    getDailyJobClass().getDeclaredConstructor(Map.class).newInstance(jobDataMap),
+                                    dateExpression);
+                            logger.debug("Scheduled astro job-startup-{} with interval of {} seconds for thing {}",
+                                    typeId, thingConfig.getInterval(), thingUid);
 
                             if (isPositionalChannelLinked()) {
                                 // positional intervalJob
-                                jobDataMap = new JobDataMap();
+                                jobDataMap = new HashMap<>();
                                 jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
                                 jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-positional");
 
-                                Date start = new Date(System.currentTimeMillis() + (thingConfig.getInterval()) * 1000);
-                                trigger = newTrigger().withIdentity("trigger-positional", thingUid).startAt(start)
-                                        .withSchedule(simpleSchedule().repeatForever()
-                                                .withIntervalInSeconds(thingConfig.getInterval()))
-                                        .build();
-                                jobDetail = newJob(PositionalJob.class).withIdentity("job-positional", thingUid)
-                                        .usingJobData(jobDataMap).build();
-                                quartzScheduler.scheduleJob(jobDetail, trigger);
+                                dateExpression = new DateExpression(
+                                        AbstractBaseJob.ISO8601_FORMAT.format(start.getTime()));
+                                getScheduler().schedule(new PositionalJob(jobDataMap), dateExpression);
                                 logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
                                         thingConfig.getInterval(), thingUid);
                             }
                         }
                     }
-                } catch (SchedulerException ex) {
+                } catch (Exception ex) {
```

Comment:
@maggu2810 Thanks for your review. I can change the code to catch each type of exception here. I have to catch six different types of exeptions here. That's the case why I use `Exception` in the first place.